app/controllers/auth_key_pair_cloud_controller.rb

```diff
@@ -112,21 +112,16 @@ def create
       ext_management_system = find_by_id_filtered(ManageIQ::Providers::CloudManager, options[:ems_id])
       kls = kls.class_by_ems(ext_management_system)
       if kls.is_available?(:create_key_pair, ext_management_system, options)
-        begin
-          kls.create_key_pair(ext_management_system, options)
-          add_flash(_("Creating %{model} %{name}") % {
-            :model => ui_lookup(:table => 'auth_key_pair_cloud'),
-            :name  => options[:name]})
-        rescue => ex
-          add_flash(_("Unable to create %{model} %{name}. %{error}") % {
-            :model => ui_lookup(:table => 'auth_key_pair_cloud'),
-            :name  => options[:name],
-            :error => get_error_message_from_fog(ex.to_s)}, :error)
+        task_id = kls.create_key_pair_queue(session[:userid], ext_management_system, options)
+
+        add_flash(_("Key Pair creation failed: Task start failed: ID [%{id}]") %
+                  {:id => task_id.to_s}, :error) unless task_id.kind_of?(Integer)
+
+        if @flash_array
+          javascript_flash(:spinner_off => true)
+        else
+          initiate_wait_for_task(:task_id => task_id, :action => "create_finished")
```

Comment:
Also, can't we have just one `if` instead of two? I mean:
```ruby
if task_id.kind_of?(Integer) # or some other valid condition
  initiate_wait_for_task()
else
  add_flash()
  javascript_flash(:spinner_off => true)
end
```